inception-active-learning/src/main/java/de/tudarmstadt/ukp/inception/active/learning/sidebar/ActiveLearningSidebar.java

```diff
@@ -606,15 +606,15 @@ private void actionRemoveHistoryItem(AjaxRequestTarget aTarget, LearningRecord a
                         null));
                 confirmationDialog.show(aTarget);
                 confirmationDialog
-                    .setConfirmAction(t -> deleteAnnotationByHistory(t, aRecord, aJcas));
+                    .setConfirmAction(t -> deleteAnnotationByHistory(t, aRecord));
             }
         }
     }
 
-    private void deleteAnnotationByHistory(AjaxRequestTarget aTarget, LearningRecord aRecord,
-        JCas aJcas)
+    private void deleteAnnotationByHistory(AjaxRequestTarget aTarget, LearningRecord aRecord)
         throws IOException, AnnotationException
     {
+        JCas aJcas = casStorageService.readCas(aRecord.getSourceDocument(), aRecord.getUser());
```

Comment:
If a user clicks the delete button, if the action is acceptance, main editor will jump to the location of that learning history firstly. So the annotation is in the current document.

Reply:
It looks quite of fragile to me to rely on the fact that some other method causes the annotation editor to jump somewhere. If somebody thinks it might be a good idea to change/refactor the behavior when the delete button is pressed, e.g. to not jump to another document but to show what is deleted in the confirmation dialog instead, then this will break any may easily go unnoticed. Better avoid it.

I note that there is also a second call to `casStorageService.readCas` in the sidebar code. Is there any particular reason you use this low-level API and not `documentService.readAnnotationCas`? In the present setup `casStorageService` should only be used directly if there are important reasons or it - otherwise CASes should be loaded through the `DocumentService`.

Reply:
No particular reason, just because each learning record has `SourceDocument`, and `documentService.readAnnotationCas` requires `AnnotationDocument`.  `casStorageService.readCas` just needs `SourceDocument`.
I'll change them.

Reply:
@reckart By using `SpanAdapter`, do you mean writing a method similar to `deleteAnnotation` in `AnnotationDetailEditorPanel`?